### PR TITLE
[UNG-1023] rework AT messaging

### PR DESCRIPTION
**Pull request by sven-hm**
This PR includes:
1. An integration test that fails if unsolicited messages are not handled properly, based on `src/main.py`.
2. Simple function to read signal quality from modem, i.e. "Reference Signal Received Quality" and "Reference Signal Received Power", compare [reference manual](https://docs.pycom.io/gitbook/assets/Monarch-LR5110-ATCmdRefMan-rev6_noConfidential.pdf). It is used in the test.
3. Extension of the `LTE` class with ~~a queue~~ logging for unsolicited messages.

### CODE REVIEW

**Comment by thinkberg** [comment]
Anything special to prepare?

**Comment by thinkberg** [comment]
Is `error_handling.py` the latest version?

```
*** UBIRCH SIM Testkit ***
Traceback (most recent call last):
  File "main.py", line 18, in <module>
  File "/flash/lib/helpers.py", line 6, in <module>
  File "/flash/lib/modem.py", line 3, in <module>
ImportError: cannot import name COLOR_MODEM_FAIL
Pycom MicroPython 1.20.2.rc6 [v1.11-01f49f7] on 2020-02-28; GPy with ESP32
Pybytes Version: 1.3.1
```

**Comment by sven-hm** [comment]
The pycom firmware should be version 1.20.2.r2 or newer!

**Comment by sven-hm** [comment]
And all python files in `scr/` should be in sync with this branch.

File: src/lib/modem.py [hunk 1]
```diff
@@ -1,70 +1,155 @@
 import time
 from network import LTE
+from error_handling import ErrorHandler, COLOR_MODEM_FAIL
```
Comment (thinkberg) [inline src/lib/modem.py]: there is no COLOR_MODEM_FAIL in error handling!

Comment (sven-hm) [reply src/lib/modem.py]: ... should work now with the last fixup!

File: src/lib/ubirch/ubirch_sim.py [hunk 2]
```diff
@@ -274,10 +268,10 @@ def _execute(self, cmd: str) -> (bytes, str):
             cmd = cmd[0] + channel_char + cmd[2:]
 
         at_cmd = 'AT+CSIM={},"{}"'.format(len(cmd), cmd.upper())
-        result = self._send_at_cmd(at_cmd)
+        result = self.lte.send_at_cmd(at_cmd)
 
-        if result[-1] == 'OK':
-            response = result[0][7:].split(',')[1]
+        if result is not None:
```
Comment (sven-hm) [inline src/lib/ubirch/ubirch_sim.py]: So if `result is None` in the Exception below `result` is not subscriptable! Compare Waldi's logs:
```
(2021.01.30 08:03:16) Traceback (most recent call last):
  File "main.py", line 245, in <module>
  File "/flash/lib/helpers.py", line 54, in send_backend_data
  File "/flash/lib/ubirch/ubirch_sim.py", line 181, in deinit
  File "/flash/lib/ubirch/ubirch_sim.py", line 178, in deinit
  File "/flash/lib/ubirch/ubirch_sim.py", line 242, in _close_channel
  File "/flash/lib/ubirch/ubirch_sim.py", line 281, in _execute
TypeError: 'NoneType' object isn't subscriptable
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Ubirch Testkit - Changelog
+
+## [1.2.0] - 2021-03-31
+### Added
+- This changelog and a patch level in versioning.
+- Handling of unsolicited messages from modem as needed with the newer pycom firmware.
+
+### Changed
+- Update pycom firmware to version v20.2.r4.
```
Comment (gruenwaldi) [inline CHANGELOG.md]: Here is a mistake in the version. should be v1.20.2.r4
Also we changed the modem firmware, didn't we? This does not show up

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+### Fixed
+- Rework of modem communication module.
+
+### Known issues
+- When network coverage is low the modem gets quite busy leading to flaws in the communication between modem and esp. A solution to this problem could be to position the device in a place with better network coverage.
+
+## [1.1] - 2020-07-06
+### Added
+- First version of Ubirch Testkit application, which works on a GPy and a PySense or PyTrack from PyCom together with a NB-IoT SIM card with connectivity from 1nce. This application can be used to test the "Ubirch protocol" in general, and espacially to test "Ubirch on a SIM". You can build your own ubirch application based on this.
```
Comment (gruenwaldi) [inline CHANGELOG.md:19]: we should add a version of the pycom firmware and the modem firmware here as well

**Review by gruenwaldi — CHANGES_REQUESTED** [review]
Please adapt proposed changes and additions

File: CHANGELOG.md (hunk 3 above)
Comment (sven-hm) [reply CHANGELOG.md]: Ok, fixed the version and added a note on the modem firmware version.

File: CHANGELOG.md (hunk 4 above)
Comment (sven-hm) [reply CHANGELOG.md:19]: Ok, added a note about pycom firmware and modem firmware versions.

**Review by gruenwaldi — APPROVED** [review]

**Pull request merged**